src/prelude/internals.mo

```diff
@@ -448,140 +448,138 @@ func @call_raw(p : Principal, m : Text, a : Blob) : async Blob {
 type @Node = { expire : [var Nat64]; id : Nat; delay : ?Nat64; job : () -> async (); pre : ?@Node; post : ?@Node };
 var @timers : ?@Node = null;
 func @prune(n : ?@Node) : ?@Node = switch n {
-    case null null;
-    case (?n) {
-        if (n.expire[0] == 0) {
-            @prune(n.post) // by corollary
-        } else {
-            ?{ n with pre = @prune(n.pre); post = @prune(n.post) }
-        }
+  case null null;
+  case (?n) {
+    if (n.expire[0] == 0) {
+      @prune(n.post) // by corollary
+    } else {
+      ?{ n with pre = @prune(n.pre); post = @prune(n.post) }
     }
+  }
 };
 
 func @nextExpiration(n : ?@Node) : Nat64 = switch n {
-    case null 0;
-    case (?n) {
-        var exp = @nextExpiration(n.pre); // TODO: use the corollary for expire == 0
-        if (exp == 0) {
-            exp := n.expire[0];
-            if (exp == 0) {
-                exp := @nextExpiration(n.post)
-            }
-        };
-        exp
-    }
+  case null 0;
+  case (?n) {
+    var exp = @nextExpiration(n.pre); // TODO: use the corollary for expire == 0
+    if (exp == 0) {
+      exp := n.expire[0];
+      if (exp == 0) {
+        exp := @nextExpiration(n.post)
+      }
+    };
+    exp
+  }
 };
 
 // Function called by backend to run eligible timed actions.
 // DO NOT RENAME without modifying compilation.
 func @timer_helper() : async () {
```

Comment:
```suggestion
func @timer_helper() : async () {
```
It would be nice if the helper took the global_timer_set capability as an argument and used it, just as we are expecting clients to do, but I guess that can happen in a separate PR if necessary

Reply:
internal are under out control, I don't see why we shouldn't use the most performant way to implement this.